pkg/network/setup/netstat.go

```diff
@@ -168,7 +168,16 @@ func ifacesStatusFromGuestAgent(vmiIfacesStatus []v1.VirtualMachineInstanceNetwo
 	for _, guestAgentInterface := range guestAgentInterfaces {
 		if vmiIfaceStatus := netvmispec.LookupInterfaceStatusByMac(vmiIfacesStatus, guestAgentInterface.Mac); vmiIfaceStatus != nil {
 			updateVMIIfaceStatusWithGuestAgentData(vmiIfaceStatus, guestAgentInterface, vmiInterfacesSpecByName)
-			vmiIfaceStatus.InfoSource = netvmispec.InfoSourceDomainAndGA
+			// For backward compatability with older virt-launchers, apply this logic:
```

Comment:
In case the `name` is there, cannot you check the `intefaceName` is there as well?
Anything reported from the GA should have `interfaceName` isn't it?
Maybe it is even enough to check just the `interfaceName` then the `name` can be dropped.

Reply:
I think you are right. We can use `interfaceName` to determine if it came from the GA or not.
There should be no case where this field is not populated when originated for the agent.

I will re-do this, thanks!